### PR TITLE
Enable checkpointing integration tests

### PR DESCRIPTION
**Pull request by SurbhiJainUSC**
# Description
This PR enables `checkpointing_test` integration test that was previously disabled in [https://github.com/AI-Hypercomputer/maxtext/pull/1482].

*Notice 1:* Once all tests pass, the "pull ready" label will automatically be assigned.
This label is used for administrative purposes. Please do not add it manually.

*Notice 2:* For external contributions, our settings currently require an approval from a MaxText maintainer to trigger CI tests.

# Tests

Integration tests pass for both TPU and GPU.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

**Review by aireenmei — APPROVED** [review]
Thanks for working on this!

File: MaxText/tests/integration_tests/checkpointing_test.py [hunk 1]
```diff
@@ -14,46 +14,107 @@
...
+  with (
+      open(metrics_file_saved, "rt", encoding="utf8") as saved,
+      open(metrics_file_restored, "rt", encoding="utf8") as restored,
+  ):
+    saved_loss = json.loads(saved.readlines()[-1])[target]
+    restored_loss = json.loads(restored.readlines()[0])[target]
+    # Checks that checkpoint restore was successful by comparing loss of last
+    # step in saved checkpoint to loss of first step in restored checkpoint
+    print("saved loss: ", saved_loss)
+    print("restored loss: ", restored_loss)
```
Comment (bvandermoon) [inline MaxText/tests/integration_tests/checkpointing_test.py:74]: These are intended to be print statements instead of maxtext logs, right?

**Review by gobbleturk — APPROVED** [review]

File: MaxText/tests/integration_tests/checkpointing_test.py [hunk 2]
```diff
@@ -14,46 +14,107 @@
...
+
 from datetime import datetime
-import subprocess
+import json
+from math import isclose
 import os.path
 import pytest
 from MaxText.globals import PKG_DIR
-from MaxText.tests.globals import TEST_DISABLE_SUBPROCESS, TEST_DISABLE_SUBPROCESS_STR
+from MaxText.train import main as train_main
```
Comment (bvandermoon) [inline MaxText/tests/integration_tests/checkpointing_test.py:31]: Not for this PR, but it would be nice if we could clean up this import in the future to something more like:

```
from MaxText.train import train
```

File: MaxText/tests/integration_tests/checkpointing_test.py [hunk 3]
```diff
@@ -14,46 +14,107 @@
...
   ]
+  train_main(
+      get_checkpointing_command(
+          run_date,
+          hardware=hardware,
+          steps=5,
+          metrics_file="saved_metrics.txt",
+          attention_type=attention_type,
+      )
+      + grain_command
```
Comment (bvandermoon) [inline MaxText/tests/integration_tests/checkpointing_test.py:94]: Why do we need grain for the checkpointing tests?

File: MaxText/tests/integration_tests/checkpointing_test.py [hunk 4]
```diff
@@ -14,46 +14,107 @@
...
+  model_params = [
+      "base_emb_dim=384",
+      "base_num_query_heads=8",
+      "base_num_kv_heads=8",
+      "base_mlp_dim=192",
+      "base_num_decoder_layers=8",
+      "head_dim=128",
+  ]
+  return [
+      None,
```
Comment (bvandermoon) [inline MaxText/tests/integration_tests/checkpointing_test.py:44]: How is `None` being used here?

Comment (SurbhiJainUSC) [reply MaxText/tests/integration_tests/checkpointing_test.py:44]: This argument is basically the module to run, like `MaxText.train`, which is not important to specify for tests.

File: MaxText/tests/integration_tests/checkpointing_test.py (hunk 1 above)
Comment (SurbhiJainUSC) [reply MaxText/tests/integration_tests/checkpointing_test.py:74]: These print statements are helpful in case the test fails and will only be present when tast fails.

File: MaxText/tests/integration_tests/checkpointing_test.py (hunk 3 above)
Comment (SurbhiJainUSC) [reply MaxText/tests/integration_tests/checkpointing_test.py:94]: This is how tests were set up before. This PR only removes `subprocess` and enables those tests back.

**Review by bvandermoon — APPROVED** [review]
